nf_core/sync.py

```diff
@@ -239,60 +243,29 @@ def push_template_branch(self):
 
-    def make_pull_request(self):
-        """Create a pull request to a base branch (default: dev),
-        from a head branch (default: TEMPLATE)
-
-        Returns: An instance of class requests.Response
-        """
-        # Check that we know the github username and repo name
-        try:
-            assert self.gh_username is not None
-            assert self.gh_repo is not None
-        except AssertionError:
-            raise PullRequestException("Could not find GitHub username and repo name")
-
-        # If we've been asked to make a PR, check that we have the credentials
-        try:
-            assert os.environ.get("GITHUB_AUTH_TOKEN", "") != ""
-        except AssertionError:
-            raise PullRequestException(
-                "Environment variable GITHUB_AUTH_TOKEN not set - cannot make PR\n"
-                "Make a PR at the following URL:\n  https://github.com/{}/compare/{}...TEMPLATE".format(
-                    self.gh_repo, self.original_branch
-                )
-            )
-
-        log.info("Submitting a pull request via the GitHub API")
-
-        pr_title = "Important! Template update for nf-core/tools v{}".format(nf_core.__version__)
-        pr_body_text = (
-            "A new release of the main template in nf-core/tools has just been released. "
-            "This automated pull-request attempts to apply the relevant updates to this pipeline.\n\n"
-            "Please make sure to merge this pull-request as soon as possible. "
-            "Once complete, make a new minor release of your pipeline. "
-            "For instructions on how to merge this PR, please see "
-            "[https://nf-co.re/developers/sync](https://nf-co.re/developers/sync#merging-automated-prs).\n\n"
-            "For more information about this release of [nf-core/tools](https://github.com/nf-core/tools), "
-            "please see the [nf-core/tools v{tag} release page](https://github.com/nf-core/tools/releases/tag/{tag})."
-        ).format(tag=nf_core.__version__)
-
-        # Try to update an existing pull-request
-        if self.update_existing_pull_request(pr_title, pr_body_text) is False:
-            # None found - make a new pull-request
-            self.submit_pull_request(pr_title, pr_body_text)
-
-    def update_existing_pull_request(self, pr_title, pr_body_text):
-        """
-        List existing pull-requests between TEMPLATE and self.from_branch
-
-        If one is found, attempt to update it with a new title and body text
-        If none are found, return False
+    def close_open_template_merge_pull_requests(self):
+        """Get all template merging branches (starting with 'nf-core-template-merge-')
+        and check for any open PRs from these branches to the self.from_branch
+        If open PRs are found, add a comment and close them
         """
         assert os.environ.get("GITHUB_AUTH_TOKEN", "") != ""
```

Comment:
This isn't surrounded by a `try` / `except` block so it actually raised an exception for a lot of use cases...

Reply:
Thanks!